src/core/main.c

```diff
@@ -228,10 +228,12 @@ _noreturn_ static void freeze_or_exit_or_reboot(void) {
         freeze();
 }
 
-_noreturn_ static void crash(int sig) {
+_noreturn_ static void crash(int sig, siginfo_t *siginfo, void *context) {
         struct sigaction sa;
         pid_t pid;
 
+        assert(siginfo);
```

Comment:
is this guaranteed to be set? If not, maybe check instead and skip the new calls if it's NULL?

Reply:
When delivering a signal with a SA_SIGINFO handler, the kernel
       does not always provide meaningful values for all of the fields
       of the siginfo_t that are relevant for that signal.

Reply:
Given it's informational-only, let's play it safe, and wrap the new code in `if (siginfo)` instead of the assert

Reply:
I checked the man page, but it does not guarantee that `siginfo` is always provided. It is safer to do as @bluca commented.

Reply:
New version force pushed, PLAL. @bluca